server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -236,28 +249,28 @@ static class Monitoring implements Runnable, Scheduler.Cancellable {
         private final ClusterService clusterService;
         private final Client client;
 
-        private final AtomicReference<String> lastSeenHealthNode;
         private final List<HealthTracker<?>> healthTrackers;
 
+        private final AtomicReference<Boolean> inFlightRequest;
```

Comment:
This is a stylistic question (at least I think it is). Any reason to use AtomicReference<Boolean> rather than AtomicBoolean? My understanding is that compareAndSet will use reference equality, which is fine here, since auto-boxed booleans objects will have == and .equals that behave the same. But auto-boxing in general is a bit confusing, so if it's possible to use AtomicBoolean, that seems a bit clearer.

Reply:
Good catch! Was indeed not intentional, so will update 👍🏻.